app/models/metric/purging.rb

```diff
@@ -85,9 +93,35 @@ def self.purge_hourly(older_than, window = nil, total_limit = nil, &block)
   end
 
   def self.purge_realtime(older_than, window = nil, total_limit = nil, &block)
-    purge_by_date(older_than, "realtime", window, total_limit, &block)
+    truncate_child_tables(older_than)
   end
 
+  # truncate metrics child tables
+  # Determines hours not being preserved and truncates them
+  # Used for realtime metrics.
+  def self.truncate_child_tables(older_than)
+    target_hours = determine_target_hours(older_than, Time.now.utc)
+    return if target_hours.blank?
```

Comment:
lets add validations for these settings
if this is blank, let the user know there is a problem

Reply:
To clarify:  `lib/vmdb/config/validator.rb` is what we can do to check this (ignore the line that this comment is on)